core-rust/state-manager/src/state_manager.rs

```diff
@@ -934,8 +963,13 @@ where
     S: QueryableProofStore + QueryableTransactionStore,
 {
     pub fn commit(&'db mut self, commit_request: CommitRequest) -> Result<(), CommitError> {
+        let commit_transactions_len = commit_request.transaction_payloads.len();
+        if commit_transactions_len == 0 {
+            return Ok(());
```

Comment:
we should never try to commit 0 txns, I wonder if this should be an error instead?